imap/mailbox.c

```diff
@@ -5715,104 +5698,18 @@ static int _mailbox_index_repack(struct mailbox *mailbox,
-
-                if (mbtype == MBTYPE_EMAIL && !mbname_isdeleted(mbname)) {
-                    // attempt to find an existing message with the same guid
-                    // and use its internaldate instead
-                    struct find_dup_rock frock = {
-                        foldernum, record->uid, { 0, UTIME_OMIT }
-                    };
-                    char guid[2*MESSAGE_GUID_SIZE+1];
-
-                    strcpy(guid, message_guid_encode(&record->guid));
-
-                    // ignore errors, it's OK for this to fail
-                    conversations_guid_foreach(cstate, guid,
-                                               find_dup_msg, &frock);
-
-                    // if we found a matching message, use its internaldate
-                    if (frock.internaldate.tv_nsec != UTIME_OMIT) {
-                        copyrecord.internaldate = frock.internaldate;
-
-                        // "remove" existing record crc from existing synccrc
-                        repack->crcs.basic ^= crc_basic(mailbox, record);
-
-                        // update existing record with new internaldate
-                        // and "add" new record crc to existing synccrc
-                        struct index_record newcrcrec = *record;
-                        newcrcrec.internaldate = frock.internaldate;
-                        repack->crcs.basic ^= crc_basic(mailbox, &newcrcrec);
-                    }
-                    else {
-                        // make sure we don't have a JMAP ID (internaldate) clash
-                        conversations_adjust_internaldate(cstate, guid,
-                                                          &copyrecord.internaldate);
-                    }
-                }
-
-                if (!dryrun) {
-                    // track this record so we can set the file timestamps later
-                    // we only need UID, internaldate, and internal_flags
-                    struct index_record *trecord =
-                        xzmalloc(sizeof(struct index_record));
-
-                    trecord->uid = copyrecord.uid;
-                    trecord->internaldate = copyrecord.internaldate;
-                    trecord->internal_flags = copyrecord.internal_flags;
-                    ptrarray_append(records, trecord);
-                }
-
-                // update G & J records
-                r = mailbox_update_conversations(mailbox, record, &copyrecord);
-                if (r) goto done;
-
-                if (mailbox->i.minor_version < 20) {
-                    // add virtual annotation to old crc
-                    buf_reset(&buf);
-                    buf_printf(&buf, UINT64_FMT, copyrecord.internaldate.tv_nsec);
-                    repack->crcs.annot ^=
-                        crc_annot(record->uid,
-                                  IMAP_ANNOT_NS "internaldate.nsec", "", &buf);
-                }
+                copyrecord.internaldate.tv_nsec = nsec;
```

Comment:
If the annotation contained a 0, we will end up with tv_nsec 0.  Later, under "calculate nanoseconds for internaldate if not yet present or re-calculating" comment, we only compute a new tv_nsec if we saw UTIME_OMIT.

So:

1. is there any place where we set tv_nsec on unannotated messages to (say) UTIME_OMIT or 0?  
2. should we set/leave it OMIT if the annotation contains 0?